src/main/java/org/truffleruby/core/string/StringOperations.java

```diff
@@ -138,13 +138,10 @@ public static String decodeUTF8(DynamicObject string) {
     }
 
     /**
-     * Create a byte[] from a CharSequence assuming a raw/ISO-8859-1 encoding
-     *
-     * @param s the CharSequence to convert
-     * @return a byte[]
+     * Create a byte[] from a String assuming a raw/ISO-8859-1 encoding
      */
     @TruffleBoundary
-    public static byte[] plain(CharSequence s) {
+    public static byte[] plain(String s) {
         // Taken from org.jruby.util.ByteList.plain
 
         if (s instanceof String) {
```

Comment:
This `instanceof` can be removed now :)
But I am not convinced `encode()` is any better than the loop below, maybe do a quick test in interpreter and with Graal?
It could compile without the encode(), and not do any allocation beyond the byte[].